### ISSUE TITLE
@order.vendor_total(vendor) returns incorrect amount if some items are taxable

### ISSUE THREAD

**Issue by johndavid400** [spree/spree_multi_vendor#87]
tl;dr - When checking out with 2 items from different vendors and 1 item is taxable, the vendor totals both include the tax for the taxable item.

I have 2 items in my cart, each one $500. The first item is in a tax_category that applies tax based on zone. The other item has no tax category assigned. I am checking out with an address that is in the zone assigned to the tax_category (state based 6%) of the first item. The Spree order total is correct as it only applies tax to the taxable item. Both items have free shipping and the order total is $1,030. So far so good.

But when calling @order.vendor_total(vendor), each vendor shows that their vendor_total is $530 for an incorrect order total of $1,060.  I expect one to be $530 and the other $500. 

I believe this happens because of the method definition is taking the order total (with tax) and subtracting the pre-tax totals of the other vendor's items in the cart combined with the pre-tax shipment totals of other vendor's items - a calculation which leaves the entire order tax amount to be added onto every vendor_total calculation:

```
def vendor_total(vendor)
  // the current approach removes all other vendor's pre-tax items/shipments
  // from the tax-included order total and returns the result - which leaves the 
  // order tax to be added to each vendor separately (problem)
  total -
  line_items.not_for_vendor(vendor).sum(:pre_tax_amount) - 
  shipments.not_for_vendor(vendor).sum(:pre_tax_amount)
end
```

I think we should be doing something more like this to calculate vendor_total:

```
def vendor_total(vendor)
  // this approach adds up the items/shipments belonging to the vendor with tax if applicable
  line_items.for_vendor(vendor).sum(:pre_tax_amount) +
  line_items.for_vendor(vendor).sum(:additional_tax_total) + 
  shipments.for_vendor(vendor).sum(:cost)
end
```

Has anyone else run into this issue or is there some logic behind the method definition that I am missing? Will be happy to submit a PR if someone else confirms this to be an issue.

Thanks,
JD

**Comment by damianlegawiec** [comment]
fixed in 2.0
